### PR TITLE
Add reference to GitHub action settings

### PR DESCRIPTION
**Pull request by penx**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,7 +57,10 @@ _Optional_ A pull request number, only required if triggered from a workflow_dis
 ## Usage
 
 Configure this action in your workflows providing the inputs described above.
-Note that this action requires a GitHub token with additional permissions. You must use the [`permissions`](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#permissions) tag to specify the required rules or configure your [GitHub account](https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/).
+Note that this action requires a GitHub token with additional permissions. You must either:
+
+1. Use the [`permissions`](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#permissions) tag to specify the required rules and under "Settings > Actions > General" check "Allow GitHub Actions to create and approve pull requests".
+2. Use a custom token [GitHub account](https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/) with these permissions configured.
```
Comment (simoneb) [inline README.md:63]: am I mistaken or this is not the setting you were referring to though?

Comment (penx) [reply README.md:63]: My understanding of the instruction before this change was that you had to either

1. Use the [`permissions`](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#permissions) tag to specify the required rules.
2. Use a custom token [GitHub account](https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/) with these permissions configured.

I extracted these to the list items above and then added

> and under "Settings > Actions > General" check "Allow GitHub Actions to create and approve pull requests".

to option 1.

This is the setting I referred to in #359

**Comment by simoneb** [comment]
@penx thanks and apologies for being slow at getting back to this. In principle I'm not against this change, but I'd like to understand better how GitHub behaves in this regard, so I'd really appreciate if you could check and clarify this, as it's not clear from the docs.

There are two things at play here: the `permissions` in the workflow yaml and the repository setting [Preventing GitHub Actions from creating or approving pull requests](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#preventing-github-actions-from-creating-or-approving-pull-requests).

I would like to understand how they interact with each other before we make this change. Here are my questions and doubts:

- This setting is in the "Workflow permissions" section of the repository configuration. Because there's another setting in that same section and because that setting can be overwritten by using `permissions` in the workflow yaml, are we sure that those same permissions aren't also overriding the setting `Preventing GitHub Actions from creating or approving pull requests`?
- Assuming that answer to the above is _no_, if we check this setting's checkbox, do we still need the permissions in the workflow yaml?

In other words, I'd like to understand if both permissions **and** checking this setting are needed, or only one of them.
